### ISSUE TITLE
Sensitive to initial value?

### ISSUE THREAD

**Issue by kimren227** [czq142857/BSP-NET-pytorch#5]
Hi Zhiqin,

I really liked your work and thanks for the awesome code. 

I'm trying to expand BSPNet to point cloud, but training becomes unstable and sensitive to changes in "p_dim", "c_dim" and torch random seeds. 

The first training iteration sometimes generates an empty or solid shape and kills all the gradients. Probably due to clamping operations in the generator. 

Don't know if you faced similar issues before, any suggestions are really appreciated.

Best,

Daxuan

**Comment by czq142857** [comment]
Hi Daxuan,

I have only tested on voxels (as inputs), and I rarely encountered any convergence issues. When I encountered a convergence problem, a re-training was usually enough. Maybe it is due to the different properties of CNN and pointcloud encoders.

I do not have an immediate solution for that. If you think clamping was the problem, you could try a "leaky" clamp akin to leaky ReLU, and see how it works.

Best,
Zhiqin

**Comment by kimren227** [comment]
Hi Zhiqin,

Thanks a lot for the suggestion. 
I have implemented a leaky clamp based on LeakyReLu and the issue seems to be resolved,

Thanks again for the valuable suggestion.

Daxuan
